src/design-system/InputDate.tsx

```diff
@@ -109,7 +108,7 @@ const InputDate: React.FC<Props> = (props) => {
         }
         tileClassName={props.tileClassName}
         tileContent={props.tileContent}
-        value={inputValue}
+        value={props.valueEntered}
```

Comment:
No, the new component does not replace `defaultValue`. I have just removed the defaultValue functionality since it wasn't being used.

Would we need `defaultValue` in the future? I'm pushing for never using `defaultValue` in the future, since `defaultValue` has some behaviors that are easy to overlook. For example, what happens when `value` is also defined? What if `value` is the empty string?

I think it makes more sense for people who want `defaultValue` behavior in the future to build it themselves using `useState` and `useEffect` so that the complexity of the behavior is apparent, and so it is easier to make changes to the default behavior. But anyway, I don't think anyone will want to use `defaultValue`, since its behavior is very nuanced.

Reply:
Thanks for the explanation, @szhu!